server_test.go

```diff
@@ -3350,6 +3350,117 @@ func TestMaxBodySizePerRequest(t *testing.T) {
 	}
 }
 
+func TestStreamRequestBody(t *testing.T) {
+	t.Parallel()
+
+	part1 := strings.Repeat("1", 1<<10)
+	part2 := strings.Repeat("2", 1<<20-1<<10)
+
+	contentLength := len(part1) + len(part2)
+
+	s := &Server{
+		Handler: func(ctx *RequestCtx) {
+			checkReader(t, ctx.RequestBodyStream(), part1)
+			checkReader(t, ctx.RequestBodyStream(), part2)
```

Comment:
You're still just writing and reading both parts after each other. What I would like to see is like this: https://play.golang.org/p/r1s8_h52e3q
So you really only send part2 when you know part1 is already read.